### ISSUE TITLE
[Question][Hw4] Allowed expression inside `assume` function

### ISSUE THREAD

**Issue by AnHaechan** [prosyslab-classroom/is593-2022-spring#23]
In `test/verifier.h`, the `assume` function is declared as `void assume(int)`.
Since it is not declared with `void assume(bool)`,
we can write program such as
```
#include "verifier.h"
int main(){
  int x = input();
  assume(x);
  assert(x!=0);
  return 0;
}
```
Should we consider the above case, too?
Or can we safely assume that the expression inside `assume` is limited to boolean expressions.

Thanks.

**Comment by hyunsukimsokcho** [comment]
Yes, you may. Btw, the usual `bool` type is represented as `int` data type in c.

**Comment by AnHaechan** [comment]
Isn't the `bool` represented as `_Bool` type in C?
AFAIK, `bool` is preprocessed into `_Bool` and `_Bool` is translated into `i1` by clang&LLVM. 
`int` is translated into `i32` by clang&LLVM.

**Comment by hyunsukimsokcho** [comment]
I wasn’t referring to the clang implementation. Just try to run the test with `void assume(bool)` in `verifier.h` and see how it goes.

**Comment by KihongHeo** [comment]
@AnHaechan you can ignore ill-typed programs.
